### PR TITLE
fix(): remove validation for existing project ns

### PR DESCRIPTION
**Pull request by priyank-upadhyay**
# Description

Support existing namespace for project creation.

Fixes #

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- Unit testing

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [x] I have added all the required unit test cases.

## Does this PR introduce a breaking change for other components like worker-operator?

```release-note

```

### CODE REVIEW

File: service/namespace_service.go [hunk 1]
```diff
@@ -84,6 +85,34 @@ func (n *NamespaceService) ReconcileProjectNamespace(ctx context.Context, namesp
 				"object_kind": metricKindNamespace,
 			},
 		)
+	} else {
+		// check if the namespace has the correct labels
+		if !util.CompareLabels(nsResource.Labels, n.getResourceLabel(namespace, owner)) {
+			nsResource.Labels = n.getResourceLabel(namespace, owner)
+			err := util.UpdateResource(ctx, nsResource)
+			nsResource.Namespace = ControllerNamespace
```
Comment (bharath-avesha) [inline service/namespace_service.go:93]: Why is this needed? A namespace object is not namespace scoped. The update will fail if you try to update the namespace field.

File: service/namespace_service.go [hunk 2]
```diff
@@ -84,6 +85,34 @@ func (n *NamespaceService) ReconcileProjectNamespace(ctx context.Context, namesp
 				"object_kind": metricKindNamespace,
 			},
 		)
+	} else {
+		// check if the namespace has the correct labels
+		if !util.CompareLabels(nsResource.Labels, n.getResourceLabel(namespace, owner)) {
+			nsResource.Labels = n.getResourceLabel(namespace, owner)
```
Comment (bharath-avesha) [inline service/namespace_service.go:91]: I think this should not be a replace operation. The cluster admin might have their own labels for whatever reason. We should only add our labels instead of replacing all labels on the ns.

Comment (priyank-upadhyay) [reply service/namespace_service.go:91]: I agree. Fixing this

File: service/namespace_service.go (hunk 1 above)
Comment (priyank-upadhyay) [reply service/namespace_service.go:93]: util.UpdateResource has a custom logic to handle both namespace and cluster scoped objects. It would handle namespace update as well.

**Review by bharath-avesha — APPROVED** [review]
